tools/run_tests/xds_k8s_test_driver/tests/api_listener_test.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+import logging
+from typing import List, Optional
+
+from absl import flags
+from absl.testing import absltest
+
+from framework import xds_k8s_testcase
+from framework.helpers import retryers
+from framework.infrastructure import k8s
+from framework.test_app import server_app
+
+logger = logging.getLogger(__name__)
+flags.adopt_module_key_flags(xds_k8s_testcase)
+
+# Type aliases
+_XdsTestServer = xds_k8s_testcase.XdsTestServer
+_XdsTestClient = xds_k8s_testcase.XdsTestClient
+
+_TD_CONFIG_RETRY_WAIT_SEC = 2
+
+
+class ApiListenerTest(xds_k8s_testcase.RegularXdsKubernetesTestCase):
+    previous_route_config_version: str
```

Comment:
This is only used by one method, it's better to make it a local var to avoid impression this is represents any kind of class state.